projects/output-and-error-handling/client/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	// Connect to the server and getting a response
+	resp, err := http.Get("http://localhost:8080")
+	// Show error message if connection is not established
+	if err != nil {
+		handleError()
+	}
+
+	// Close the response body after been fully read
+	defer resp.Body.Close()
+
+	// Read response body and store it in a var
+	body, err := io.ReadAll(resp.Body)
+
+	// Show error message if we cannot read the response body
+	if err != nil {
+		handleError()
+	}
+	// Convert response body from binary to string
+	sb := string(body)
+
+	// Handle cases depending on the Status code of the response
+	switch resp.StatusCode {
+	case 200:
+		fmt.Fprint(os.Stdout, sb+"\n")
+	case 429:
+		handleRateLimited(resp.Header.Get("Retry-After"))
+	case 500:
+		fmt.Fprint(os.Stderr, sb+"\n")
```

Comment:
If response is 500, body might be empty (most of the time!). So this won't produce a very useful error log